From e0cdda117ba11fc1f0b75d5bf012769deb50dcf8 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 23 Nov 2023 10:25:49 +0100
Subject: [PATCH 0073/1695] drivers: clk: clk_print_tree() stub for disabled
 CFG_DRIVERS_CLK

Add an stub implementation of clk_print_tree() when CFG_DRIVERS_CLK
is disabled. This allows generic code as the statistics PTA to not
rely on #ifdef CFG_DRIVERS_CLK pragma.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e50e62c55456083317e32c1932a821760aace826 drivers: clk: clk_print_tree() stub for disabled CFG_DRIVERS_CLK]
Change-Id: I6892954fabd69c37ee9d17d34d1c30fad3ea73f4
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from f729983620e8c6d6470091d9db0e261c25d1d852]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/include/drivers/clk.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/include/drivers/clk.h b/core/include/drivers/clk.h
index 0c77353c4..340a79790 100644
--- a/core/include/drivers/clk.h
+++ b/core/include/drivers/clk.h
@@ -200,6 +200,11 @@ TEE_Result clk_get_rates_array(struct clk *clk, size_t start_index,
 			       unsigned long *rates, size_t *nb_elts);
 
 /* Print current clock tree summary on output console (info trace level) */
+#ifdef CFG_DRIVERS_CLK
 void clk_print_tree(void);
-
+#else
+static inline void clk_print_tree(void)
+{
+}
+#endif /* CFG_DRIVERS_CLK */
 #endif /* __DRIVERS_CLK_H */
-- 
2.47.2

